ClimsoftVer4/ClimsoftVer4/ucrFormDaily2.vb

```diff
@@ -256,33 +256,39 @@ Public Class ucrFormDaily2
         Next
     End Sub
     ''' <summary>
-    ''' Checks if the computer computed total is same as the user total
-    ''' If not gives a warning
+    ''' Checks if total for current element is required
+    ''' Checks if the computed total is same as the user entered total.
     ''' </summary>
     Public Sub checkTotal()
-        'Check total if required
-        ' Am not sure how to get this value yet
-        'If totalRequired = 1 Then
-
+        'Check total if required from obselements table from qcTotalRequired field
+        Dim clsDataDefinition As DataCall
+        Dim dtbl As DataTable
         Dim elemTotal As Integer = 0
         Dim expectedTotal As Integer
         Dim ctr As Control
         Dim ctrVFP As New ucrValueFlagPeriod
 
+        clsDataDefinition = New DataCall
+        clsDataDefinition.SetTableName("obselements")
+        clsDataDefinition.SetFields(New List(Of String)({"qcTotalRequired"}))
+        clsDataDefinition.SetFilter("elementId", "=", ElementId, bIsPositiveCondition:=True, bForceValuesAsString:=False)
 
-        expectedTotal = Val(ucrInputTotal.GetValue)
-
-        For Each ctr In Me.Controls
-            If TypeOf ctr Is ucrValueFlagPeriod Then
-                ctrVFP = ctr
-                elemTotal = elemTotal + Val(ctrVFP.ucrValue.GetValue)
+        dtbl = clsDataDefinition.GetDataTable()
+        If dtbl IsNot Nothing AndAlso dtbl.Rows.Count > 0 Then
+            If dtbl.Rows(0).Item("qcTotalRequired") = 1 Then
+                expectedTotal = Val(ucrInputTotal.GetValue)
+                For Each ctr In Me.Controls
+                    If TypeOf ctr Is ucrValueFlagPeriod Then
+                        ctrVFP = ctr
+                        elemTotal = elemTotal + Val(ctrVFP.ucrValue.GetValue)
+                    End If
+                Next
+                If elemTotal <> expectedTotal Then
+                    MessageBox.Show("Value in [Total] textbox is different from that calculated by computer!", caption:="Error in total")
+                    ucrInputTotal.txtBox.Focus()
+                    ucrInputTotal.txtBox.BackColor = Color.Cyan
+                End If
```

Comment:
Instead of 

```
ucrInputTotal.txtBox.Focus()
ucrInputTotal.txtBox.BackColor = Color.Cyan
```

Do 

```
ucrInputTotal.GetFocus()
ucrInputTotal.SetBackColor(Color.Cyan)
```

Reply:
Sounds good :-)

Reply:
Nice